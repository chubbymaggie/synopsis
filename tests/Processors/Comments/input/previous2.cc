// -*- Comments.SSFilter -*-
// -*- Comments.Previous -*-

class Foo
{
public:
  int first;  //< first comment
  int second; //< second comment
private:
  int third;  //< third comment
};
