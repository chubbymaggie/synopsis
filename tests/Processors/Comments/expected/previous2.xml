<?xml version='1.0' encoding='ISO-8859-1'?>
<ast>
 <declarations>
  <instance accessibility="0" class="Synopsis.AST.Class" file="Processors/Comments/input/previous2.cc" line="4" name="Foo" type="class">
   <declarations>
    <item>
     <instance accessibility="1" class="Synopsis.AST.Variable" constr="0" file="Processors/Comments/input/previous2.cc" line="7" name="Foo.first" type="data member">
      <vtype>
       <instance class="Synopsis.Type.Base" language="C++" name="int"/>
      </vtype>
      <annotations>
       <key>
        comments
       </key>
       <value/>
      </annotations>
     </instance>
    </item>
    <item>
     <instance accessibility="1" class="Synopsis.AST.Variable" constr="0" file="Processors/Comments/input/previous2.cc" line="8" name="Foo.second" type="data member">
      <vtype/>
      <annotations>
       <key>
        comments
       </key>
       <value>
        <item>
          second comment
        </item>
       </value>
      </annotations>
     </instance>
    </item>
    <item>
     <instance accessibility="1" class="Synopsis.AST.Builtin" file="Processors/Comments/input/previous2.cc" line="9" name="EOS" type="EOS">
      <annotations>
       <key>
        comments
       </key>
       <value>
        <item>
         &lt; second comment
        </item>
       </value>
      </annotations>
     </instance>
    </item>
    <item>
     <instance accessibility="3" class="Synopsis.AST.Variable" constr="0" file="Processors/Comments/input/previous2.cc" line="10" name="Foo.third" type="data member">
      <vtype/>
      <annotations>
       <key>
        comments
       </key>
       <value/>
      </annotations>
     </instance>
    </item>
    <item>
     <instance accessibility="3" class="Synopsis.AST.Builtin" file="Processors/Comments/input/previous2.cc" line="10" name="EOS" type="EOS">
      <annotations>
       <key>
        comments
       </key>
       <value>
        <item>
         &lt; third comment
        </item>
       </value>
      </annotations>
     </instance>
    </item>
   </declarations>
   <annotations>
    <key>
     comments
    </key>
    <value>
     <item>
      -*- Comments.SSFilter -*-
-*- Comments.Previous -*-
     </item>
     <item/>
    </value>
   </annotations>
  </instance>
 </declarations>
 <types>
  <instance class="Synopsis.Type.Declared" language="C++" name="EOS">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="Foo">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="Foo.first">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="Foo.second">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="Foo.third">
   <declaration/>
  </instance>
 </types>
 <files>
  <instance class="Synopsis.SourceFile.SourceFile" language="C++" name="Processors/Comments/input/previous2.cc" primary="1">
   <declarations>
    <item/>
    <item/>
    <item/>
    <item/>
    <item/>
    <item/>
   </declarations>
   <includes/>
   <macro_calls/>
  </instance>
 </files>
</ast>
